### PR TITLE
[HIP][HIPIFY] Split HIP_ARRAY_DESCRIPTOR struct to HIP_ARRAY_DESCRIPTOR and HIP_ARRAY3D_DESCRIPTOR

### PR DESCRIPTION
**Pull request by emankov**
[Reason] To be compatible with CUDA [#1133]

Update HIP code, hipify-clang, tests and docs

[TODO] Add support of the corresponding functions on nvcc fallback path

P.S. This is a PR based on closed #1135

### CODE REVIEW

**Comment by emankov** [comment]
There was a single failed test:
`directed_tests/runtimeApi/memory/hipMemoryAllocateCoherentDriver.tst***Failed  Required regular expression not found.Regex=[PASSED]  0.09 sec
`, which **failed not due to the PR**, it was also failed with previous PR's, for instance: #1213.

File: src/hip_memory.cpp [hunk 1]
```diff
@@ -690,29 +694,28 @@ hipError_t hipMallocArray(hipArray** array, const hipChannelFormatDesc* desc, si
...
     *array = (hipArray*)malloc(sizeof(hipArray));
-    array[0]->type = pAllocateArray->flags;
-    array[0]->width = pAllocateArray->width;
-    array[0]->height = pAllocateArray->height;
-    array[0]->depth = pAllocateArray->depth;
-    array[0]->drvDesc = *pAllocateArray;
+    array[0]->type = pAllocateArray->Flags;
+    array[0]->width = pAllocateArray->Width;
+    array[0]->height = pAllocateArray->Height;
+    array[0]->depth = pAllocateArray->Depth;
```
Comment (ansurya) [inline src/hip_memory.cpp]: Below properties are not set and being accessed in ihipBindTextureToArrayImpl. So when 3D array is mapped to textures it fails with segmentation fault.
   array[0]->Format = pAllocateArray->Format;
    array[0]->NumChannels = pAllocateArray->NumChannels;

Comment (emankov) [reply src/hip_memory.cpp]: Thanks, fixed.

**Comment by emankov** [comment]
Ping

**Comment by emankov** [comment]
Just a gentle reminder

**Pull request merged**